t/redis.t

```diff
@@ -0,0 +1,38 @@
+use strict;
+use warnings;
+
+use Data::Chronicle::Writer;
+use RedisDB;
+use Date::Utility;
+use Test::More;
+use Test::Exception;
+require Test::NoWarnings;
+
+BEGIN {
+    plan skip_all => 'needs TEST_REDIS=redis://localhost:6379' unless $ENV{TEST_REDIS};
+}
+
+my $data = {sample => 'data'};
+
+subtest "Call Set after dropping the connection" => sub {
+
+    my $connection = RedisDB->new(url => $ENV{TEST_REDIS});
+
+    my $writer = Data::Chronicle::Writer->new(
+        publish_on_set => 1,
+        cache_writer   => $connection
+    );
+     # calling set() which will call mset() and put the flag `multi`
+    $writer->set('namespace', 'category', $data, Date::Utility->new, 0);
+
+    # Kill All Client Connections
+    my $cmd = "redis-cli -p 6379 CLIENT KILL TYPE normal";
+    for (1..5) {   system($cmd);   sleep(1);}
```

Comment:
Can we check the `kill` result here ? Make sure all connections dropped.

Reply:
`system("my commands in a string")` is unsafe, prefer `system qw(my commands in an --array)`

That said, would it be possible to use [Data::Chronicle::Mock](https://github.com/binary-com/perl-Data-Chronicle/blob/master/lib/Data/Chronicle/Mock.pm) to `->exec("CLIENT KILL")` instead?

Reply:
@zakame from what we understand, `Data::Chronicle::Mock` is mocking the redis functionality by preparing a database and adding mocking the subs with queries to insert and extract data from it.  Thus, it has no connection to actual redis and thus we cant execute `KILL` as there is none. 

> system("my commands in a string") is unsafe, prefer system qw(my commands in an --array)

ok will change

Reply:
@chylli  we cant really check the `kill` result because the `kill` returns the number of the connection that was killed, which is uncertain (we do not know how many of them exists). This is also one of the reason we execute the `kill` command multiple times, just to confirm we killed them

Reply:
Reviewing this subtest, it seems better to actually check that `Data::Chronicle::Writer` will reconnect through its underlying `cache_writer` handle rather than relying on an outside agent to kill that handle.

Reply:
@limchengyang use `client list type normal` to get all id and and kill then  check all those client ids are disappeared ?
https://redis.io/commands/client-list

Reply:
> Reviewing this subtest, it seems better to actually check that `Data::Chronicle::Writer` will reconnect through its underlying `cache_writer` handle rather than relying on an outside agent to kill that handle.

We need the connection to be dropped by the remote server to have the same issue, not sure how reconnect is related to this case?

@chylli I think more than 0 is enough to check other than that a fixed number is not reliable